mirror-index.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -e
+
+INDEX_IMG=""
+INDEX_MIRROR=""
+MIRROR_MANIFEST_FLD="./mirror-index-manifests/"
+function parse_arguments() {
+    if [[ $# -lt 1 ]]; then print_usage; exit; fi
+
+    while [[ $# -gt 0 ]]; do
+        key="$1"
+        case $key in
+            -i|--index)
+            INDEX_IMG=$2
+            shift 2
+            ;;
+            -t|--target)
+            INDEX_MIRROR=$2
+            shift 2
+            ;;
+            -v|--verbose)
+            set -x
+            shift 1
+            ;;
+            '--help')
+            print_usage
+            exit 0
+            ;;
+            *)
+            echo -e "Unknown option $1 is specified. See usage:\n"
+            print_usage
+            exit 0
+        esac
+    done
+}
+
+print_usage () {
+  echo "Usage:   $0 -i [ORIGINAL_INDEX] -t [TARGET_INDEX]"
+  echo "Options:
+  --index,  -i     the index which should be tested
+                   To find index see Index Image Location in cvp-test-report of
+                   web-terminal-operator-metadata-container-${WANTED_VERSION}
+                   http://external-ci-coldstorage.datahub.redhat.com/cvp/cvp-redhat-operator-bundle-image-validation-test/
+
+  --target, -t     the target where mirrored index should be pushed
+
+  --verbose, -v    print the verbose output during execution
+
+  --help,   -h     help
+  "
+}
+
+parse_arguments "$@"
+
+echo "Preparing mirrored index $INDEX_MIRROR for $INDEX_IMG"
+
+rm -rf $MIRROR_MANIFEST_FLD/
+
+# Go through $INDEX_IMG and grab all the mappings
+docker pull $INDEX_IMG
+
+oc adm catalog mirror --insecure=true --manifests-only --to-manifests=$MIRROR_MANIFEST_FLD $INDEX_IMG quay.io/wto
+
+cd $MIRROR_MANIFEST_FLD
+
+# Create the new mapped-images.txt that contains only the web-terminal mappings
+grep web-terminal mapping.txt > mapped-images.txt
+
+# cut chars after : to evaluate index repo
+INDEX_REPO=$(echo $INDEX_IMG | sed -e 's/^\(.*\):.*$/\1/')
```

Comment:
Could be simplified to
```suggestion
INDEX_REPO=${INDEX_IMG%:*}
```